lms/djangoapps/edraak_certificates/generator.py

```diff
@@ -0,0 +1,323 @@
+
+            if next_width >= max_width:
+                yield de_reverse(line).strip()
+                line = next_word
+            else:
+                line += u' ' + next_word.strip()
+
+        if line:
+            yield de_reverse(line).strip()
+
+    def save(self):
+        self.ctx.showPage()
+        self.ctx.save()
+
+    def course_org_disclaimer(self):
+        if self.course_org == 'MITX':
+            return self._("A course of study offered by Edraak with cooperation from MITx. "
+                          "The learning experience has been supervised and managed by the course team.")
+        else:
+            return self._("A course of study offered by Edraak. The learning experience has been supervised and "
+                          "managed by the course team.")
+
+    def generate_and_save(self):
+        self.init_context()
+
+        x = 10.8
+        self.add_certificate_bg()
+        self.add_course_org_logo(self.course_org, self.course_id)
+
+        self.draw_bidi_text(self._("This is to certify that:"), x, 5.8, size=0.25)
+
+        user_profile_size = 0.42 if contains_rtl_text(self.user_profile_name) else 0.5
+        self.draw_single_line_bidi_text(self.user_profile_name, x, 5.124, size=user_profile_size, bold=True)
+
+        self.draw_bidi_text(self._("Successfully completed:"), x, 4.63, size=0.25)
+
+        course_name_size = 0.31 if contains_rtl_text(self.course_name) else 0.33
+
+        sponsor = get_course_sponsor(self.course_id)
+        if sponsor:
+            self.draw_single_line_bidi_text(self.course_name, x, 4.1, size=course_name_size, bold=True)
+            self.draw_bidi_text(self._("This course is sponsored by:"), x, 3.5, size=0.25)
+            self.add_course_sponsor_logo(sponsor, self.course_id)
+        else:
+            self.draw_bidi_text(self.course_name, x, 4.1, size=course_name_size, bold=True)
+            if not self.is_english_course():
+                self.draw_bidi_text(self.course_desc, x, 3.74, size=0.16)
+            else:
+                self.draw_english_text(self.course_desc, x, 3.74, size=0.16)
+
+        date_x = 2.01
+
+        words = self._("Course{new_line}Certificate{new_line}of Completion").split('{new_line}')
+
+        for idx, word in enumerate(words):
+            font_size = 0.27
+            line_height = font_size * 1.3
+            y = 6.1 - (idx * line_height)
+
+            self.draw_bidi_center_text(word, date_x, y, size=font_size, bold=True)
+
+        self.draw_single_line_bidi_text(self.instructor, x, 1.8, size=0.26, bold=True)
+
+        if not self.is_english_course():
+            self.draw_bidi_text(self.course_org_disclaimer(), x, 1.44, size=0.16)
+        else:
+            self.draw_english_text(self.course_org_disclaimer(), x, 1.44, size=0.16)
+
+        self.draw_bidi_center_text(self.course_end_date, date_x, 4.82, size=0.27)
+
+        self.save()
+
+
+def generate_certificate(request, course):
```

Comment:
I hate to do this, but when an edX function requires it, I have no other solution.